### PR TITLE
updating control plugin

### PR DESCRIPTION
**Pull request by PabloEspejeLS**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### CODE REVIEW

File: hunter_base/config/hardware_controllers.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+controller_manager:
+  ros__parameters:
+    update_rate: 50
+    use_sim_time: false
+
+    diff_drive_controller:
+      type: hunter_controller/HunterController
+    joint_state_broadcaster:
+      type: joint_state_broadcaster/JointStateBroadcaster
```
Comment (marc-hanheide) [inline hunter_base/config/hardware_controllers.yaml:9]: Is the joint_state_broadcaster declared as a run dependency in the package.xml?

File: hunter_base/hunter_base.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<library path="hunter_base">
+  <class name="hunter_base/DiffBotSystemHardware"
+         type="hunter_base::DiffBotSystemHardware"
+         base_class_type="hardware_interface::SystemInterface">
+    <description>
```
Comment (marc-hanheide) [inline hunter_base/hunter_base.xml:5]: is hardware_interface declared a run dependency? Overall, please check that all run dependencies are declared.

File: hunter_base/launch/hunter.launch.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    robot_state_pub_node = Node(
+        package="robot_state_publisher",
+        executable="robot_state_publisher",
+        output="both",
+        parameters=[robot_description],
+        remappings=[
+            ("/diff_drive_controller/cmd_vel_unstamped", "/cmd_vel"),
+        ],
+    )
+    # rviz_node = Node(
```
Comment (marc-hanheide) [inline hunter_base/launch/hunter.launch.py]: the argument for "gui" is still in but this is commented out?

File: hunter_base/launch/hunter.launch.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    #     executable="rviz2",
+    #     name="rviz2",
+    #     output="log",
+    #     arguments=["-d", rviz_config_file],
+    #     condition=IfCondition(gui),
+    # )
+
+    joint_state_broadcaster_spawner = Node(
+        package="controller_manager",
+        executable="spawner",
```
Comment (marc-hanheide) [inline hunter_base/launch/hunter.launch.py:69]: is controller manager declared?

**Review by marc-hanheide — COMMENTED** [review]
General question: I like you made the launch part of the package in principle, but it does change what was provided upstream. @ibrahimhroob do we want this or keep this fork clean of additions? If we keep it we must make sure that the run/exec dependencies match what is required to launch it.

**Comment by ibrahimhroob** [comment]
I am not sure about this PR, is there another way to add those changes?

**Comment by ibrahimhroob** [comment]
@PabloEspejeLS can you please submit another PR with minimal changes? by that I mean with the things you need to add only and please try to make them as small as possible and it would be great to add comments on why you have introduced those changes.

**Comment by hariharan20** [comment]
Hi @ibrahimhroob, There are many changes to be done on this repo further to this commits. Could you please merge this changes for now. I will create future pull requests with comments on the changes. Thanks

**Comment by ibrahimhroob** [comment]
Hi @hariharan20, done as requested, but my main concern is if the main repo got updated, it may cause conflict issues when sync with the main stream repo

**Pull request merged**